Added refactoring description of parse() in omitValue.java

Samuelhertzberg opened the pull request:
Added a refactoring description of the method parse(). This is only one of two alternatives. The other being just splitting the function. However I felt that writing two methods would just be to much. Should resolve #29

Samuelhertzberg committed:
Added refactoring desc

Added a refactoring description of the function parse. Via this method 
of refactoring we could reduce the complexity by about 66% Far greater 
than the required 35% mark.

francislaus reviewed (changes requested):
Good job, but you should change "However there is also smarter ways to acquire the correct string" to "However there **are** also smarter ways to acquire the correct string"

Samuelhertzberg committed:
Rewrote the refactor description

I realized that the hashmap was not going to work. So I changed it to 
this method instead. Should finish up #29

Samuelhertzberg commented:
Did a complete rewrite @francislaus

Samuelhertzberg committed:
Type

Changed type as per Franz request

francislaus reviewed (approved):
Looks good
